src/main/java/org/sagebionetworks/bridge/services/HealthDataEx3Service.java

```diff
@@ -50,13 +81,37 @@ public void deleteRecordsForHealthCode(String healthCode) {
         healthDataEx3Dao.deleteRecordsForHealthCode(healthCode);
     }
 
-    /** Retrieves the record for the given ID. */
-    public Optional<HealthDataRecordEx3> getRecord(String id) {
+    /** Retrieves the record for the given ID. Provide the download link if requested. */
+    public Optional<HealthDataRecordEx3> getRecord(String id, boolean download) {
         if (StringUtils.isBlank(id)) {
             throw new BadRequestException("ID must be specified");
         }
 
-        return healthDataEx3Dao.getRecord(id);
+        Optional<HealthDataRecordEx3> record = Optional.ofNullable(healthDataEx3Dao.getRecord(id))
+                .orElseThrow(() -> new EntityNotFoundException(HealthDataRecordEx3.class));
+
+        if (download) {
+            record.get().setDownloadUrl(generatePresignedUrl(record.get(), GET).toExternalForm());
+        }
+        return record;
+    }
+
+    /**
+     * Returns the url path of the given record.
+     * @param record the record
+     * @return the url path of the given record.
+     */
+    private URL generatePresignedUrl(HealthDataRecordEx3 record, HttpMethod method) {
+        long expiration = DateTime.now().plusMinutes(EXPIRATION_IN_MINUTES).getMillis();
+        record.setDownloadExpiration(expiration);
+
+        S3Key = UploadUtil.getRawS3KeyForUpload(record.getAppId(), record.getStudyId(),
+                this.uploadService.getUpload(record.getId()), record);
```

Comment:
Digging through the code, it looks like we no longer use record.getStudyId(). Instead, we get the studyId from the Participant, and the health record can be exported to multiple studies.

For now, just assume that the uploaded data will always be under the app and ignore the study for creating the S3Key.

Similarly, we can remove studyId from UploadUtils and from the expected download URL in the tests.